cmd/kosli/attestSonar.go

```diff
@@ -156,6 +164,7 @@ func newAttestSonarCmd(out io.Writer) *cobra.Command {
 	cmd.Flags().StringVar(&o.projectKey, "sonar-project-key", "", sonarProjectKeyFlag)
 	cmd.Flags().StringVar(&o.serverURL, "sonar-server-url", "https://sonarcloud.io", sonarServerURLFlag)
 	cmd.Flags().StringVar(&o.revision, "sonar-revision", o.commitSHA, sonarRevisionFlag)
+	cmd.Flags().IntVar(&o.maxRetries, "max-retries", 0, sonarMaxRetriesFlag)
```

Comment:
would it be better to default the max-retries to something like 3?
if the objective is to get data eventually if it is not available yet, then it is probably more user friendly to retry by default rather than fail first and then make the user specify a number of retries.

On another note, in this context, it might be better to have a wait flag instead of retry. You wait (by retrying) until the results are available or a certain timeout is reached. The user will never know how many retries should work, but might have a reasonable guess on how long they want to wait (timeout).

Reply:
we have maxAPIRetries flag for kosli api requests. I suppose for other async 3rd part interactions, the need would be similar to sonar (keep waiting for results to become available or timeout).